validator_instance.go

```diff
@@ -656,6 +661,34 @@ func (v *Validate) VarCtx(ctx context.Context, field interface{}, tag string) (e
 	return
 }
 
+// VarCtxMap validates a single variable (in a map) using tag style validation and allows passing of contextual
+// validation information via context.Context. This allow usage of "Field" validations via ValidateMap() by using
+// map lookup format "eqfield=[AnotherMapKey] AnotherMapKeyValue" and "required_if=[AnotherMapKey] AnotherMapKeyValue"
+//
+// It returns InvalidValidationError for bad values passed in and nil or ValidationErrors as error otherwise.
+// You will need to assert the error if it's not nil eg. err.(validator.ValidationErrors) to access the array of errors.
+// validate Array, Slice and maps fields which may contain more than one error
+func (v *Validate) VarCtxMap(ctx context.Context, parent any, field interface{}, tag string) (err error) {
+	if len(tag) == 0 || tag == skipValidationTag {
+		return nil
+	}
+
+	ctag := v.fetchCacheTag(tag)
+
+	val := reflect.ValueOf(field)
+	vd := v.pool.Get().(*validate)
+	vd.top = val
+	vd.isPartial = false
+	vd.traverseField(ctx, reflect.ValueOf(parent), val, vd.ns[0:0], vd.actualNs[0:0], defaultCField, ctag)
```

Comment:
this is the only difference from `VarCtx` (2nd arg of `reflect.ValueOf(parent)`) - I found it cleaner to have a new func for it, but fine with any other feedback to this y'all have